src/crimson/admin/admin_socket.cc

```diff
@@ -0,0 +1,896 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
+// vim: ts=8 sw=2 smarttab
+/*
+ * Ceph - scalable distributed file system
+ *
+ * Copyright (C) 2011 New Dream Network
+ *
+ * This is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License version 2.1, as published by the Free Software
+ * Foundation.  See file COPYING.
+ *
+ */
+
+#include "common/version.h"
+#include "crimson/net/Socket.h"
+// activate for some inline unit-testing: #define UNIT_TEST_OSD_ASOK
```

Comment:
For merge, it would be nice to create an actual unit test to run in the usual make check paths.

Reply:
will do.